go.mod

```diff
@@ -9,6 +9,7 @@ require (
 	github.com/klauspost/compress v1.15.0
 	github.com/mdp/qrterminal/v3 v3.0.0
 	github.com/spf13/cobra v1.5.0
+	gitlab.com/yawning/nyquist.git v0.0.0-20211010104234-909e7e222b8f
```

Comment:
I we confident to use this repo? Seems like this is on low usage / activity repo. 
Also go.dev suggest another repo of the same person if I am correct: https://pkg.go.dev/search?q=nyquist&m=

Was this repo library analysed to use: https://pkg.go.dev/github.com/flynn/noise?

Reply:
In general it would be good to carefully assess and discuss new external libraries inclusion as it might raise certain security risks and can lead to compromising whole solution.

Reply:
Addition some comments from internal discussion:
**flynn/noise**

- less code
- less external dependencies
- seems to be more popular and active (29 closed PRs from many contributors vs. one person project)

**yawning/nyquist**

- better structure
- slightly higher coverage (85.4% vs 83%, basically similar)
- author is known for contribution to Tor and low-level cryptographic primitives

Alternative: 
(wireguard-go case) mostly we use one of the protocol from the noise family, we could think of implementing that as a module within the project later and get rid of the library at some point later. This helps in keeping the dependencies low

Reply:
> Wasm module is not building up:
> 
> ```
> ➜  wormhole-william git:(piegames/noise-transit) GOOS=js GOARCH=wasm go build -o ../../dist/wormhole.wasm ./wasm/module
> # gitlab.com/yawning/x448.git/internal/toolchain
> ../../../../../go/pkg/mod/gitlab.com/yawning/x448.git@v0.0.0-20211010103852-8630503de309/internal/toolchain/constraints.go:45:6: undefined: __SOFTWARE_REQUIRES_GOARCH_NOT_WASM__

As a general rule, my more recent cryptography packages explicitly do not support WASM, as it is impossible to guarantee constant time operation where required.  This is a WASM problem.  There was a draft extension somewhere that proposed adding timing guarantees to certain opcodes but AFAIK the issue is largely ignored instead.